### PR TITLE
feat(connector): [Airwallex] Currency Unit Conversion

### LINKED ISSUE #2215: [FEATURE]: [Airwallex] Currency Unit Conversion
### :memo: Feature Description

- Each currency can be described in terms of base or minor units. 
- For instance, when dealing with USD currency, the Cent is considered as the minor unit, while the Dollar is considered the base unit. 
- In Hyperswitch, the amount value is expected to be always provided in minor units. 
- For example: In USD, If the amount is 1 in base unit (Dollar), then it will be equivalent to 100 in minor units (Cent)
- Some of the connectors integrated require the amount to be converted to a desirable unit before being passed to them. 
- We have functions `to_currency_base_unit` and `to_currency_lower_unit` in place to convert the minor unit amount to its decimal equivalent and vice versa. These conversions are handled explicitly based on the connector.

### :hammer: Possible Implementation

- ConnectorCommon trait have been implemented for the connector. 
- This trait contains `get_currency_unit` method. This method needs to be implemented. 
- It will define what type of conversion needs to be done during `connector_router_data` creation. Concurrently handle the creation of `connector_router_data` to be passed for the connector request body creation.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/2196

:bookmark: Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by SagarDevAchar**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Addressing Issue: #2215
- Modified two files in `hyperswitch/crates/router/src/connector/`
  - `airwallex.rs`
    - Implement `get_currency_unit` function
    - Modify `ConnectorIntegration` implementations for `Airwallex`
  - `airwallex/transformers.rs`
    - Implement `AirwallexRouterData<T>` structure and functionality
    - Modify implementation for `AirwallexPaymentsRequest`

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Closes #2215.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/airwallex/transformers.rs [hunk 1]
```diff
@@ -538,17 +575,19 @@ pub struct AirwallexRefundRequest {
...
-    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
+    fn try_from(
+        item: &AirwallexRouterData<&types::RefundsRouterData<F>>,
+    ) -> Result<Self, Self::Error> {
         Ok(Self {
             request_id: Uuid::new_v4().to_string(),
             amount: Some(utils::to_currency_base_unit(
-                item.request.refund_amount,
-                item.request.currency,
+                item.router_data.request.refund_amount,
```
Comment (srujanchikke) [inline crates/router/src/connector/airwallex/transformers.rs:586]:
```
amount : Some(item.amount) 
```

Comment (SagarDevAchar) [reply crates/router/src/connector/airwallex/transformers.rs]: Updated!

Comment (SagarDevAchar) [reply crates/router/src/connector/airwallex/transformers.rs]: Thanks @pixincreate!

**Review by srujanchikke — APPROVED** [review]
Looks good to me!

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @SagarDevAchar!

**Comment by swangi-kumari** [comment]
Hey @SagarDevAchar ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Pull request merged**